src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts

```diff
@@ -598,4 +599,53 @@ describe('ExerciseDetailsStudentActionsComponent', () => {
         expect(window.alert).toHaveBeenCalledWith('artemisApp.exercise.maxAthenaResultsReached');
         expect(result).toBeFalse();
     });
+
+    it('start theia button should be visible when profile is active and url is set', () => {
```

Comment:
```suggestion
    it.each([
        { theiaUrl: undefined, activeProfiles: [], expectedVisibility: false},
        { ...},
        { ... }
    ])('start theia button should visible=$expectedVisibility if url=$theiaUrl and profiles=$activeProfiles', ({ theiaUrl, activeProfiles, expectedVisibility}) => {
```
I think you could combine the three tests into a parametrised one to reduce duplication.

The syntax should be somewhat similar to the suggestion.

Reply:
Very good idea, I didn´t know about `it.each(..)` :)